### PR TITLE
listener: tiny refactor

### PR DESCRIPTION
**Pull request by lambdai**
Flush them to avoid tackle this in a huge PR

### CODE REVIEW

File: pilot/pkg/model/sidecar.go [hunk 1]
```diff
@@ -447,20 +447,12 @@ func (sc *SidecarScope) GetEgressListenerForRDS(port int, bind string) *IstioEgr
 
 // Services returns the list of services imported by this egress listener
 func (ilw *IstioEgressListenerWrapper) Services() []*Service {
-	if ilw == nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar.go:450]: Why donot keep this to prevent developer misuse?

File: pilot/pkg/model/sidecar.go [hunk 2]
```diff
@@ -606,42 +602,27 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 	return filteredServices
 }
 
-func matchingServices(importedHosts []host.Name, service *Service, ilw *IstioEgressListenerWrapper) []*Service {
+func matchingServices(importedHosts []host.Name, service *Service, ilw *IstioEgressListenerWrapper) *Service {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar.go:605]: nit: matchingService

File: pilot/pkg/model/sidecar.go [hunk 3]
```diff
@@ -606,42 +602,27 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
...
-			if portMatched {
-				for _, port := range service.Ports {
-					if port.Port == int(ilw.IstioListener.Port.GetNumber()) {
-						ports := []*Port{}
 						sc := service.DeepCopy()
-						ports = append(ports, port)
-						sc.Ports = ports
-						importedServices = append(importedServices, sc)
-						break
+						sc.Ports = []*Port{port}
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar.go:618]: @ramaraochavali please take a look

**Comment by ramaraochavali** [comment]
LGTM apart from the comments by @hzxuzhonghu

File: pilot/pkg/model/sidecar.go (hunk 1 above)
Comment (lambdai) [reply pilot/pkg/model/sidecar.go:450]: I checked the contexts of IstioEgressListenerWrapper::Services(). There will be an immediate access of the other members of `IstioEgressListenerWrapper`. 
The production code is not ready for a nil value IstioEgressListenerWrapper, we can add it back if nil value does help in some flow.

**Pull request merged**